src/services/ui/src/features/package-actions/IssueRai.tsx

```diff
@@ -0,0 +1,110 @@
+import { Alert } from "@/components";
+import * as SC from "@/features/package-actions/shared-components";
+import { useParams } from "react-router-dom";
+import { z } from "zod";
+import { Info } from "lucide-react";
+import { getUser } from "@/api/useGetUser";
+import { Authority } from "shared-types";
+import { unflatten } from "flat";
+import { zAttachmentOptional, zAttachmentRequired } from "@/pages/form/zod";
+import { submit } from "@/api/submissionService";
+
+type Attachments = keyof z.infer<typeof issueRaiSchema>["attachments"];
+export const issueRaiSchema = z.object({
+  additionalInformation: z.string(),
+  attachments: z.object({
+    formalRaiLetter: zAttachmentRequired({ min: 1 }),
+    other: zAttachmentOptional,
+  }),
+});
+
+export const issueRaiDefaultAction: SC.ActionFunction = async ({
+  request,
+  params,
+}) => {
+  try {
+    const formData = Object.fromEntries(await request.formData());
+
+    const unflattenedFormData = unflatten(formData);
+
+    const data = issueRaiSchema.parse(unflattenedFormData);
+
+    const user = await getUser();
+    const authority = Authority["1915b"];
+    await submit({
+      data: { ...data, id: params.id },
+      endpoint: "/action/issue-rai",
+      user,
+      authority,
+    });
+
+    return {
+      submitted: true,
+    };
+  } catch (err) {
+    console.log(err);
+    return { submitted: false };
+  }
+};
+
+export const IssueRai = () => {
+  const { handleSubmit } = SC.useSubmitForm();
+  const { id } = useParams();
+
+  SC.useDisplaySubmissionAlert(
+    "RAI issued",
+    `The RAI for ${id} has been submitted. An email confirmation will be sent to you and the state.`
+  );
+
+  return (
+    <>
+      <SC.Heading title="1915(b) Waiver Formal RAI Details" />
```

Comment:
I think it's worth refactoring the other action forms to use this method of abstraction! Not in this pr, but let's ticket that if we like this pattern. Cc: @benjaminpaige @mdial89f

Reply:
honestly, all forms could utilize it. It unifies the styling which is one area our approach to making form pages leaves open for variance right now.